### PR TITLE
Add badge for new content in courses

### LINKED ISSUE #4792: "New"/"Updated" indicators for course modules
Allow for a "New" badge to distinguish newly added modules in web.dev Courses, and "Updated" badge for updated modules.

### PR DESCRIPTION
**Pull request by devnook**
Fixes #4792

### CODE REVIEW

**Comment by robdodson** [comment]
This looks good. I sent a quick email to Jiwoong and Una/Adam to get their thoughts. Folks may request that we add the word `New` to the badge, but that's the only thing that popped to mind for me.

**Comment by devnook** [comment]
I added NEW keyword and updated dot, as per design discussion.

File: src/styles/components/_drawer-course.scss [hunk 1]
```diff
@@ -82,6 +82,19 @@
...
+
+    &::after {
+      display: inline-block;
+      height: 0.5em;
+      margin-left: 0.6em;
+    }
+
+    &[data-badge='new']::after {
+      color: $WEB_PRIMARY_COLOR;
+      content: 'NEW';
```
Comment (robdodson) [inline src/styles/components/_drawer-course.scss:94]: Could we use https://developer.mozilla.org/en-US/docs/Web/CSS/attr() to set the content here? Maybe the markup would look like:

```
<a data-new data-new-label="{{ 'i18n.courses.new' | i18n(locale) }}">
```

File: test/unit/src/site/_filters/is-new-content.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const assert = require('assert');
```
Comment (robdodson) [inline test/unit/src/site/_filters/is-new-content.js:1]: yay tests!

File: src/site/_data/i18n/courses.yml [hunk 3]
```diff
@@ -3,3 +3,6 @@ learn_css:
     en: Learn CSS!
   description:
     en: "Let’s gleefully breakdown complex aspects of CSS into digestible episodes covering everything from accessibility to z-index."
+
+new:
+  en: new
```
Comment (robdodson) [inline src/site/_data/i18n/courses.yml:8]: maybe we should put this in i18n.common since it seems like something we may use elsewhere on the site

**Comment by robdodson** [comment]
From a UX perspective, it feels like the New badge should be hidden after I visit a module.

image

Could we update the css so if something is completed by the user it hides the new badge?

File: src/styles/components/_drawer-course.scss (hunk 1 above)
Comment (devnook) [reply src/styles/components/_drawer-course.scss]: done

File: src/site/_data/i18n/courses.yml (hunk 3 above)
Comment (devnook) [reply src/site/_data/i18n/courses.yml]: done

**Comment by devnook** [comment]
PTAL

**Review by robdodson — APPROVED** [review]

**Review by robdodson — CHANGES_REQUESTED** [review]
There was a merge conflict so I pulled the PR down to rebase it and now I'm noticing some small things that need fixing.

**Comment by robdodson** [comment]
Ok this started as a small change and become something bigger 😅  Initially I was just doing a rebase and wanted to fix the background color on the links, but then I noticed there was a lot of FOUC because an element will render the `NEW` badge and then remove it when the completed attribute is set. It looked a bit odd so I decided to see if we could fix it by inlining the element.

So the changes are:

- Adds an `inlinePageScripts` njk property. This lets a layout inline custom elements if they need to render to avoid FOUC
- Refactors the course-links element so it doesn't have any dependencies. Instead we inline it so it can set attributes on its children during initial render and avoid FOUC.
- Adds analytics to course-links so it will fire for every 10% that a user has completed.
- Adds a course key to the course_progress localstorage so it can support multiple courses.

**Review by MichaelSolati — APPROVED** [review]

**Review by devnook — COMMENTED** [review]
lgtm

**Pull request merged**